lib/galaxy/tools/actions/upload_common.py

```diff
@@ -120,11 +120,14 @@ def persist_uploads(params, trans):
                                                    local_filename=local_filename)
             elif type(f) == dict and 'local_filename' not in f:
                 raise Exception('Uploaded file was encoded in a way not understood by Galaxy.')
-            if 'url_paste' in upload_dataset and upload_dataset['url_paste'] and upload_dataset['url_paste'].strip() != '':
-                upload_dataset['url_paste'] = datatypes.sniff.stream_to_file(
-                    StringIO(validate_url(upload_dataset['url_paste'], trans.app.config.fetch_url_whitelist_ips)),
```

Comment:
Maybe just adding a `strip()` to this line instead would be the best. So the `else` branch is triggered the same as before

Reply:
We can, of course. Very minor issue though, IMHO: we're calling strip() twice. It doesn't matter for performance or readability, but if down the road we need to do something else to that value, we'll have to remember to call strip (a third time). If the pasted url should have no whitespace, I'd trim it before doing any kind of processing. But that's a very minor consideration; besides, there's the added line as a downside.. Up to you.

Reply:
In your current implementation if you get `  ` you return ``, in the previous one you'll return `None`. Not sure how important it is in this case but upload can to be pretty finicky.

Reply:
sounds very important! checking....

Reply:
@martenson what do you think of this approach? I split it in two: (a) getting the value if it exists, and (b) processing that value. I think this is slightly more readable, but I'm torn. If you think it's too much, let's go with what you suggested - just calling strip() again.
But if this does work, i'll add a couple unit tests.

Reply:
Are you still sure you'd choose it over this?

```diff
-                    StringIO(validate_url(upload_dataset['url_paste'], trans.app.config.fetch_url_whitelist_ips)),
+                    StringIO(validate_url(upload_dataset['url_paste'].strip(), trans.app.config.fetch_url_whitelist_ips)),
```

Reply:
> if down the road we need to do something else to that value

That url value disappears in either case, it is either None, or a file handle.

Reply:
It does disappear.. And there's really no place we would reuse that function in this file. So yes, I think you're right, let's keep it simple. Will submit in a min.

Reply:
Done.